components/ShakeComponent.js

```diff
@@ -0,0 +1,242 @@
+import React, { useEffect, useState } from "react"
+import { Accelerometer } from "expo-sensors"
+import { View, Text, StyleSheet, Image, Vibration } from "react-native"
+import AppLoading from "expo-app-loading"
+import {
+    useFonts,
+    WorkSans_100Thin,
+    WorkSans_200ExtraLight,
+    WorkSans_300Light,
+    WorkSans_400Regular,
+    WorkSans_500Medium,
+    WorkSans_600SemiBold,
+    WorkSans_700Bold,
+    WorkSans_800ExtraBold,
+    WorkSans_900Black,
+    WorkSans_100Thin_Italic,
+    WorkSans_200ExtraLight_Italic,
+    WorkSans_300Light_Italic,
+    WorkSans_400Regular_Italic,
+    WorkSans_500Medium_Italic,
+    WorkSans_600SemiBold_Italic,
+    WorkSans_700Bold_Italic,
+    WorkSans_800ExtraBold_Italic,
+    WorkSans_900Black_Italic,
+} from "@expo-google-fonts/work-sans"
+
+import Loader from "./Loader"
+import Header from "./Header"
+import Footer from "./Footer"
+
+
+const isShaking = (data) => {
+    const totalForce = Math.abs(data.x) + Math.abs(data.y) + Math.abs(data.z)
+    return totalForce > 1.78
+}
+
+const Shakecomponent = () => {
+    let [fontsLoaded] = useFonts({
+        WorkSans_100Thin,
+        WorkSans_200ExtraLight,
+        WorkSans_300Light,
+        WorkSans_400Regular,
+        WorkSans_500Medium,
+        WorkSans_600SemiBold,
+        WorkSans_700Bold,
+        WorkSans_800ExtraBold,
+        WorkSans_900Black,
+        WorkSans_100Thin_Italic,
+        WorkSans_200ExtraLight_Italic,
+        WorkSans_300Light_Italic,
+        WorkSans_400Regular_Italic,
+        WorkSans_500Medium_Italic,
+        WorkSans_600SemiBold_Italic,
+        WorkSans_700Bold_Italic,
+        WorkSans_800ExtraBold_Italic,
+        WorkSans_900Black_Italic,
+    })
+
+    Accelerometer.setUpdateInterval(400)
+    const [fetchedData, setFetchedData] = useState([])
+    const [loading, setLoading] = useState(false)
+    const [subscription, setSubscription] = useState(null)
+
+    const [data, setData] = useState({
+        x: 0,
+        y: 0,
+        z: 0,
+    })
+
+    useEffect(() => {
+        fetchData()
+    }, [])
+
+    useEffect(() => {
+        subscribe()
+        return () => unsubscribe()
+    }, [])
+
+    useEffect(() => {
+        if (isShaking(data)) {
+            fetchData()
+        }
+    }, [data])
+
+    const subscribe = () => {
+        setSubscription(
+            Accelerometer.addListener((accelerometerData) => {
+                setData(accelerometerData)
+            })
+        )
+    }
+
+    const unsubscribe = () => {
+        subscription && subscription.remove()
+        setSubscription(null)
+    }
+
+    const fetchData = () => {
+        setLoading(true)
+        fetch("https://www.boredapi.com/api/activity")
+            .then(res => res.json())
+            .then(data => setTimeout(() => setFetchedData(data), 600))
```

Comment:
Perhaps, you can add .catch error here, in case the API is not successfully fetched